### ISSUE TITLE
pito build fails on OSX 10.12.1 Sierra

### ISSUE THREAD

**Issue by iskolbin** [moai/moai-community#4]
Running pito giving me
```
Will use the /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/util/build/build-osx.sh to build /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/bin/osx/moai binary for use by pito...
User defaults from command line:
    IDEDerivedDataPathOverride = /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build

=== BUILD TARGET libmoai-osx-apple OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-sdl OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx OF PROJECT libmoai WITH CONFIGURATION Release ===
Check dependencies

=== BUILD TARGET libmoai-osx-3rdparty-core OF PROJECT libmoai WITH CONFIGURATION Release ===
Check dependencies

=== BUILD TARGET libmoai-osx-3rdparty-crypto OF PROJECT libmoai WITH CONFIGURATION Release ===
Check dependencies

=== BUILD TARGET libmoai-osx-3rdparty-sdl OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-audio-sampler OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-box2d OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-crypto OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-fmod-ex OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-http-client OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-http-server OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-image-jpg OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-image-png OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-image-pvr OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-image-webp OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-luaext OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-sim OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-untz OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-zl-core OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-zl-crypto OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-zl-vfs OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD AGGREGATE TARGET libmoai-osx-all OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

** BUILD SUCCEEDED **

User defaults from command line:
    IDEDerivedDataPathOverride = /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build

=== BUILD TARGET libmoai-osx-audio-sampler OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-3rdparty-core OF PROJECT libmoai WITH CONFIGURATION Release ===
Check dependencies

=== BUILD TARGET libmoai-osx OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-apple OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-3rdparty-crypto OF PROJECT libmoai WITH CONFIGURATION Release ===
Check dependencies

=== BUILD TARGET libmoai-osx-3rdparty-sdl OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-box2d OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-crypto OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-harfbuzz OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-http-client OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-http-server OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-image-jpg OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-image-png OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-image-pvr OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-image-webp OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-luaext OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-sdl OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-sim OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-untz OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-zl-core OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-zl-crypto OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET libmoai-osx-zl-vfs OF PROJECT libmoai WITH CONFIGURATION Release ===

Check dependencies

=== BUILD TARGET moai OF PROJECT moai-osx WITH CONFIGURATION Release ===

Check dependencies

Ld /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build/Build/Products/Release/moai normal x86_64
    cd /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/osx
    export MACOSX_DEPLOYMENT_TARGET=10.12
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -L/Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build/Build/Products/Release -F/Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build/Build/Products/Release -filelist /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build/Build/Intermediates/moai-osx.build/Release/moai.build/Objects-normal/x86_64/moai.LinkFileList -mmacosx-version-min=10.12 -Xlinker -object_path_lto -Xlinker /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build/Build/Intermediates/moai-osx.build/Release/moai.build/Objects-normal/x86_64/moai_lto.o -stdlib=libc++ -fobjc-link-runtime -lmoai-osx-3rdparty-core -lmoai-osx-3rdparty-crypto -lmoai-osx-3rdparty-sdl -lmoai-osx-apple /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build/Build/Products/Release/libmoai-osx-audio-sampler.a -lmoai-osx-box2d -lmoai-osx-crypto /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build/Build/Products/Release/libmoai-osx-harfbuzz.a -lmoai-osx-http-client -lmoai-osx-http-server /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build/Build/Products/Release/libmoai-osx-image-jpg.a /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build/Build/Products/Release/libmoai-osx-image-png.a /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build/Build/Products/Release/libmoai-osx-image-pvr.a /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build/Build/Products/Release/libmoai-osx-image-webp.a -lmoai-osx-luaext -lmoai-osx-sdl -lmoai-osx-sim -lmoai-osx-untz -lmoai-osx-zl-core -lmoai-osx-zl-crypto -lmoai-osx-zl-vfs -lmoai-osx -lstdc++ -lcrypto -lssl -lSystem -framework AudioToolbox -framework AudioUnit -framework Carbon -framework Cocoa -framework CoreAudio -framework CoreFoundation -framework CoreServices -framework CoreVideo -framework ForceFeedback -framework Foundation -framework IOKit -framework OpenGL -framework SystemConfiguration -Xlinker -dependency_info -Xlinker /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build/Build/Intermediates/moai-osx.build/Release/moai.build/Objects-normal/x86_64/moai_dependency_info.dat -o /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build/Build/Products/Release/moai
ld: library not found for -lcrypto
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** BUILD FAILED **


The following build commands failed:
	Ld /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/xcode/build/Build/Products/Release/moai normal x86_64
(1 failure)
Need to confirm local moai binary was built ...
Something went wrong with building /Users/ilyakolbin/Documents/moaisdk-osx-1.7.5/sdk/moai/util/moai .. exiting!
```
So linker cannot find crypto from openssl package I guess. Any thoughts how to fix this? Thanks in advance:)

**Comment by halfnelson** [comment]
Hrm they may have  renamed it. In theory we build our own crypto and ssl
libs, so it should be linking to them. The build script just builds the
xcode project in the sdk/moai folder. Maybe if you open and build in xcode
it will help debug.

**Comment by halfnelson** [comment]
@iskolbin  does this comment help:
https://github.com/erlio/vernemq/issues/206#issuecomment-252498082

looks like this isnt the only project that got caught out. Maybe we just need to open the xcodeproj and add that extra library path

**Comment by halfnelson** [comment]
resolved 1.7.6
